ui/src/criteria/concept.test.tsx

```diff
@@ -0,0 +1,154 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { EntityInstancesApiContext } from "apiContext";
+import { Cohort, Group, GroupKind } from "cohort";
+import { CohortUpdaterProvider } from "cohortUpdaterContext";
+import { ConceptCriteria } from "criteria/concept";
+import * as tanagra from "./tanagra-api";
+
+test.each([
+  [
+    "source",
+    {
+      instance: {
+        concept_name: {
+          stringVal: "test-concept",
+        },
+        concept_id: {
+          int64Val: 1234,
+        },
+        domain_id: {
+          int64Val: 5678,
+        },
+      },
+      matches: ["test-concept", "1234", "5678", "Source"],
+    },
+  ],
+  [
+    "standard",
+    {
+      instance: {
+        concept_name: {
+          stringVal: "test-concept",
+        },
+        concept_id: {
+          int64Val: 1234,
+        },
+        standard_concept: {
+          stringVal: "S",
+        },
+      },
+      matches: ["test-concept", "1234", "Standard"],
+    },
+  ],
+  [
+    "missing concept_id",
+    {
+      instance: {
+        concept_name: {
+          stringVal: "test-concept",
+        },
+      },
+      matches: ["0-0 of 0"],
+    },
+  ],
+])("%s", async (name, { instance, matches }) => {
+  await renderCriteria([instance]);
+
+  matches.forEach((match) => {
+    screen.getByText(match);
```

Comment:
Some of these questions may be naive, I haven't used the react test renderer library extensively:

- On the surface, this test is not making any assertions, and this line looks like a no-op. Is there some undocumented (AFAICT) side-effect of `getByText()`, e.g. that it throws an error? It may be more readable to actually assert something anyways..
- From the docs, it seems that `getByText` is an async function -is that not the case here, and if so shouldn't there be some awaits in this test?

Reply:
- getByText throws if it can't find a matching element (see the summary table at https://testing-library.com/docs/queries/about). So the full assertion could be written as `expect(getByText(match)).toBeInDocument()` but that doesn't seem to actually add any value since it's functionally equivalent. 
- getByText is the non-async version. findByText (like at line 142) is async.